h2o-py/tests/testdir_algos/glm/pyunit_GH_16312_contrained_GLM_beta_constraint_NPE_large.py

```diff
@@ -117,13 +127,29 @@ def test_bad_lambda_specification():
         "compute_p_values" : compute_p_values,
         "calc_like" : calc_like,
         "solver" : solver,
-        "linear_constraints": linear_constraints
+        "linear_constraints": linear_constraints,
+        "beta_constraints": beta_constraints
     }
 
     model = glm(**params)
     model.train(x = predictors, y = response, training_frame = train_data)
     print(model.coef())
-        
+    print(glm.getConstraintsInfo(model))
+
+    # beta constraints
+    bc = []
+    name = 'x1'
+    upper_bound = 1.5
+    bc.append([name, upper_bound])
+
+    beta_constraints2 = h2o.H2OFrame(bc)
+    beta_constraints2.set_names(["names", "upper_bounds"])
+    
+    params['beta_constraints'] = beta_constraints2
+    model = glm(**params)
+    model.train(x = predictors, y = response, training_frame = train_data)
+    print(model.coef())
+    print(glm.getConstraintsInfo(model))
```

Comment:
You wrote: no need to check anything. But what about adding an assertion at the end that the model is not None? I suggest adding a try-catch and throwing a clear message instead of letting the test fail (in case of some problem).

Reply:
If the `model` is `None` then it would fail in `model.train(...)`. What benefit would an assert have (especially if it would be at the end)? Or did you mean something like:
```diff
    model = glm(**params)
+   assert model is not None, "The model is None"
    model.train(x = predictors, y = response, training_frame = train_data)
    print(model.coef())
    print(glm.getConstraintsInfo(model))
```
?

If so, I don't understand the reason of adding the try except clause.

(Sorry if it is obvious, I haven't properly started the review yet so I might still get it.)

Reply:
The advantage of failing with an assert message is to explain why it failed. If the NPE occurs and we know the potential problem, we can explain it in the message and save time in the future. We can also catch only NPE and other errors shown as usual. 

I suggest improving the message from "The model is None" to something like "The model training failed with NPE. It could be caused by beta constraints setting."

Reply:
Without my fix, the test will fail.  However, after my fix, the NPE is no longer there.  However, I did add a test because I know the beta constraints are always satisfied.  Hence, I check that the beta constraints are always satisfied.

Reply:
From @maurever , she is not comfortable for tests without some kind of assert statement.  However, @tomasfryda feels that if the assert statement is too general, it is not very meaningful. 

Taking both of your suggestions, I added assert statements that will check and make sure that the beta constraints are satisfied at the end of the model building process.